core/import_enhanced/registry.ts

```diff
@@ -0,0 +1,34 @@
+export type Optional<T> = T | undefined;
+
+export interface Entry {
+  type: "file" | "folder";
+  value: string;
+}
+
+export interface ModContents {
+  readonly update_time: string;
+  readonly contents: Entry[];
+}
+
+export interface ModVersionMap {
+  [key: string]: ModContents;
+}
+
+export interface ModInfo {
+  readonly name: string;
+  readonly description?: string;
+  // this array should well sorted
+  // latest version should in versions[0]
+  readonly versions?: string[];
+}
+export type ModInfoList = ModInfo[];
+
+export interface Registry {
+  readonly REGISTRY_ID: string;
+  // Allow registry don't have std module mirror
+  std(): Promise<Optional<ModInfo>>;
+  stdContents(version: string[]): Promise<Optional<ModVersionMap>>;
```

Comment:
I think we shouldn't treat `std` like a special module. We should just use `modVersionList("std")`

Reply:
But If I recall correctly that in denoland, std have different api between std and x, If they're the same, just call mod one in std

Reply:
They are all the same API.

Reply:
Ok, you're right, I will change it